### PR TITLE
Fix: Lazy Load experiment caused the background not working some cases [ED-8898]

### PR DESCRIPTION
**Pull request by maork-elementor**

### CODE REVIEW

File: modules/lazyload/assets/scss/frontend.scss [hunk 1]
```diff
@@ -1,3 +1,8 @@
-body.e-lazyload .lazyloaded, body.e-lazyload .lazyloaded .elementor-background-overlay {
+.lazyloaded:not( .elementor-motion-effects-element-type-background ),
```
Comment (rotemee) [inline modules/lazyload/assets/scss/frontend.scss:1]: Please add comments that explain the selectors' targets.

File: includes/elements/container.php [hunk 2]
```diff
@@ -690,6 +690,13 @@ protected function register_background_overlay_controls() {
 							$background_overlay_selector => '--background-overlay: \'\';',
 						],
 					],
+					'image' => [
```
Comment (rotemee) [inline includes/elements/container.php:693]: @maork-elementor The background doesn't work in a container (see take for more details).

File: modules/lazyload/module.php [hunk 3]
```diff
@@ -54,6 +54,13 @@ private function update_element_attributes( Element_Base $element ) {
 
 		if ( $keys ) {
 			$background_image_url = Utils::get_array_value_by_keys( $settings, $keys );
+			if ( ! $background_image_url ) {
```
Comment (rotemee) [inline modules/lazyload/module.php:57]: Why do you have a separated condition when the $background_image_url does not exist?
why not if/else?

File: modules/lazyload/module.php [hunk 4]
```diff
@@ -54,6 +54,13 @@ private function update_element_attributes( Element_Base $element ) {
...
+				// If the background image is not set, check also for overlay image.
+				$keys = Utils::get_array_value_by_keys( $control_data, [ 'background_lazyload', 'overlay_keys' ] );
+				if ( $keys ) {
+					$background_image_url = Utils::get_array_value_by_keys( $settings, $keys );
+				}
+			}
```
Comment (rotemee) [inline modules/lazyload/module.php:63]: Add a space below the condition and consider adding spaces in general for readability (to keep on a consistent code style in the project).

File: modules/lazyload/assets/scss/frontend.scss (hunk 1 above)
Comment (maork-elementor) [reply modules/lazyload/assets/scss/frontend.scss:1]: done

File: includes/elements/container.php (hunk 2 above)
Comment (maork-elementor) [reply includes/elements/container.php:693]: @rotemee 
I just test is with container experiment ON : 

Only background image - worked.
Background image & overlay image - worked. 

Screen Shot 2022-12-01 at 9 03 56

This is the CSS for the container overlay : 

Screen Shot 2022-12-01 at 9 04 57

What do I need to do to reproduce the problem?

File: modules/lazyload/module.php (hunk 4 above)
Comment (maork-elementor) [reply modules/lazyload/module.php:63]: My PHPCS is not allowing me to add space below the condition.
The empty lines are auto removed when I save the file.
I'm using our config file `phpcs.xml`

File: includes/elements/container.php (hunk 2 above)
Comment (rotemee) [reply includes/elements/container.php:693]: @maork-elementor:
"Create a container, and set a background overlay → the background does not load."
Try only a background overlay (without a regular background).

Comment (maork-elementor) [reply includes/elements/container.php:693]: You are right, Fixed.

File: modules/lazyload/module.php (hunk 4 above)
Comment (maork-elementor) [reply modules/lazyload/module.php:63]: PHPCS error is generated by the line break in both my local and the actual PR test.

**Review by rotemee — DISMISSED** [review]

File: includes/elements/column.php [hunk 5]
```diff
@@ -298,7 +298,8 @@ protected function register_controls() {
 						'background_lazyload' => [
 							'active' => true,
 							'keys' => [ 'background_image', 'url' ],
-							'selector' => '.elementor-column-wrap',
+							'overlay_keys' => [ 'background_overlay_image', 'url' ],
```
Comment (Nevoss) [inline includes/elements/column.php:301]: the overlay is not a different control? lets talk about it

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: includes/elements/column.php (hunk 5 above)
Comment (maork-elementor) [reply includes/elements/column.php]: Keys are removed, using new logic to support multiple background controls in the same element.

**Review by Nevoss — APPROVED** [review]

**Pull request merged**